pkg/publicapi/endpoint/compute/endpoint_approve.go

```diff
@@ -20,27 +20,25 @@ import (
 //	@Failure	403	{object}	string
 //	@Failure	500	{object}	string
 //	@Router		/api/v1/compute/approve [get]
-func (s *Endpoint) approve(res http.ResponseWriter, req *http.Request) {
-	request, err := signatures.UnmarshalSigned[bidstrategy.ModerateJobRequest](req.Context(), req.Body)
+func (s *Endpoint) approve(c echo.Context) error {
+	request, err := signatures.UnmarshalSigned[bidstrategy.ModerateJobRequest](c.Request().Context(), c.Request().Body)
 	if err != nil {
-		publicapi.HTTPError(req.Context(), res, err, http.StatusBadRequest)
-		return
+		return echo.NewHTTPError(http.StatusBadRequest, err.Error())
```

Comment:
Perhaps we are not at the point where we are ready to utilize this feature yet, but we could improve this through some middleware by doing:
```go
return echo.NewHTTPError(
    http.StatusBadRequest,
    "failed to unmarshal request").
    WithInternal(err)
```
This will allow use to make assertions on the error type in middleware. It will also allow us to include the specific error in our logs message via the `Err()` method. Also has some security benefit since the `Internal` field is tagged with `json:"-"`, which means it will be ignored when the HTTPError struct is serialized to JSON. Great for preventing potentially sensitive internal error details from being exposed to our end users.

Reply:
yeah we would need to inject our own error handler for central handling of errors, which is what I like the most about echo. We need to rethink `bacerrors` package first

Reply:
We could adopt one of the existing solutions from: https://awesome-go.com/error-handling/